src/components/Faucet/Claim.tsx

```diff
@@ -54,20 +148,27 @@ export const useClaim = () => {
   useEffect(() => {
     if (!pendingClaim) return
     const url = new URL(window.location.href)
-    const provider: string | null = url.searchParams.get('provider')
+    let state: string | null = url.searchParams.get('state')
     const code: string | null = url.searchParams.get('code')
-    const recipient: string | null = url.searchParams.get('recipient')
+
+    state = atob(state || '')
+    const stateParams = JSON.parse(state || '[]')
+    const provider = stateParams.find((p: any) => p.param === 'provider')?.value
+    const recipient = stateParams.find((p: any) => p.param === 'recipient')?.value
 
     if (!code || !provider || !recipient) return
 
+    // Remove the querystring from the browser current url
+    window.history.replaceState({}, '', `${window.location.pathname}${window.location.hash}`)
+
     claimTokens(provider, code, recipient)
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [pendingClaim])
 
-  const handleSignIn = async (provider: string) => {
+  const handleSignIn = async (provider: string, recipient: string, signinUrlParams: signinUrlParams[]) => {
```

Comment:
Could you define this (and other exported methods of the hook) types and export such types? It would be a lot more useful for end developers not having to specify the entire function signature every time they use it.